internal/pkg/cli/pipeline_update.go

```diff
@@ -197,6 +210,36 @@ func (o *updatePipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) erro
 	return nil
 }
 
+func (o *updatePipelineOpts) ConnectionName() (string, error) {
```

Comment:
Is this a duplicate method of:
https://github.com/aws/copilot-cli/blob/b6bf01ff9ddbb3373a2624a29ad31d4cf14cca13/internal/pkg/deploy/pipeline.go#L171-L187

Can we think of a way of  using `deploy.BitbucketSource` instead?

Reply:
I think in `deployPipeline` we can typecast `in.Source` to be:
```
repo := `in.Source.(deploy.BitbucketSource)`
repo.ConnectionName()
```

Reply:
Oh! That's so cool--it's like reaching into the future! I wonder if we can do the same with some of the repo and owner parsing that's duplicated in `pipeline_init.go` and `deploy/pipeline.go`...?

Reply:
That'd be great :D

Reply:
I'll do it in a separate PR bc it's unrelated to polling.